libvast/src/system/index.cpp

```diff
@@ -1056,144 +1211,63 @@ index(index_actor::stateful_pointer<index_state> self,
+                       *self);
+            rp.deliver(query_cursor{query_id, 0u, 0u});
+            self->send(client, atom::done_v);
+            return;
+          }
+          auto num_candidates = detail::narrow<uint32_t>(candidates.size());
+          auto scheduled
+            = std::min(num_candidates, self->state.taste_partitions);
+          if (auto err = self->state.pending_queries.insert(
+                query_state{.query = query,
+                            .client = client,
+                            .candidate_partitions = num_candidates,
+                            .requested_partitions = scheduled},
+                std::move(candidates)))
+            rp.deliver(err);
+          rp.deliver(query_cursor{query_id, num_candidates, scheduled});
+          schedule_lookups(self->state);
+        });
       return rp;
     },
-    [self](const uuid& query_id, uint32_t num_partitions) -> caf::result<void> {
-      auto sender = self->current_sender();
-      auto client = caf::actor_cast<receiver_actor<atom::done>>(sender);
-      // Sanity checks.
-      auto iter = self->state.pending.find(query_id);
-      if (iter == self->state.pending.end()) {
-        VAST_WARN("{} drops query for unknown query id {}", *self, query_id);
-        return caf::make_error(ec::lookup_error,
-                               fmt::format("unknown query id: {}", query_id));
-      }
-      auto& query_state = iter->second;
-      if (!sender) {
-        VAST_WARN("{} ignores query {} from anonymous sender", *self, query_id);
-        self->send(self, atom::worker_v, query_state.worker);
-        return {};
-      }
-      if (num_partitions == 0) {
-        VAST_WARN("{} ignores query {} for zero partitions", *self, query_id);
-        self->send(self, atom::worker_v, query_state.worker);
-        return {};
-      }
-      auto now = std::chrono::steady_clock::now();
-      // Get partition actors, spawning new ones if needed.
-      auto actors
-        = self->state.collect_query_actors(query_state, num_partitions);
-      // Delegate to query supervisor (uses up this worker) and report
-      // query ID + some stats to the client.
-      VAST_DEBUG("{} scheduled {} more partition(s) for query id {}"
-                 " with {} partitions remaining",
-                 *self, actors.size(), query_id, query_state.partitions.size());
-      auto delta = std::chrono::steady_clock::now() - now;
-      VAST_TRACEPOINT(query_resume, query_id.as_u64().first, actors.size(),
-                      delta.count());
-      self->send(query_state.worker, atom::supervise_v, query_id,
-                 query_state.query, std::move(actors), client);
-      // Cleanup if we exhausted all candidates.
-      if (query_state.partitions.empty())
-        self->state.pending.erase(iter);
-      return {};
+    [self](atom::resolve,
+           vast::expression& expr) -> caf::result<catalog_result> {
+      auto lookup_id = vast::uuid::random();
```

Comment:
Please be consistent with the naming. Is a lookup ID a query ID?

Reply:
There is no `query` involved in this path, so it would be a bit misleading to name this variable `query_id`. This is also not an addition from this PR, it only shows up because of the merges.